os/kernel/pthread/pthread_setspecific.c

```diff
@@ -97,28 +97,17 @@
 
 int pthread_setspecific(pthread_key_t key, FAR const void *value)
 {
-	FAR struct pthread_tcb_s *rtcb = (FAR struct pthread_tcb_s *)this_task();
-	struct pthread_key_s *cur_key;
+	struct pthread_tcb_s *rtcb = (FAR struct pthread_tcb_s *)this_task();
+	struct task_group_s *group = rtcb->cmn.group;
 
-	DEBUGASSERT((rtcb->cmn.flags & TCB_FLAG_TTYPE_MASK) == TCB_FLAG_TTYPE_PTHREAD);
+	DEBUGASSERT(group && (rtcb->cmn.flags & TCB_FLAG_TTYPE_MASK) == TCB_FLAG_TTYPE_PTHREAD);
```

Comment:
Is it a special case that, we need to check for task type for PTHREAD types?   Just curious to know whether we need to have this kind of check for other APIs like, pthread_key_create, key_delete or key_destroy ?

Reply:
Task can create and delete key to use it from child pthread as gRPC does.
key_destroy is called from pthread_exit so that not needed.
Anyway key functionality is working on pthread, not task.